header.php

```diff
@@ -44,8 +44,34 @@
 			<?php endif; ?>
 		</div><!-- .site-branding -->
 
-		<nav id="site-navigation" class="main-navigation">
-			<button class="menu-toggle" aria-controls="primary-menu" aria-expanded="false"><?php esc_html_e( 'Primary Menu', '_s' ); ?></button>
+		<nav
+			id="site-navigation"
+			class="main-navigation"
+			<?php if ( _s_is_amp() ) : ?>
+				[class]=" siteNavigationMenu.expanded ? 'main-navigation toggled' : 'main-navigation' "
+			<?php endif; ?>
+		>
+			<?php if ( _s_is_amp() ) : ?>
+				<amp-state id="siteNavigationMenu">
+					<script type="application/json">
+						{
+							"expanded": false
+						}
+					</script>
+				</amp-state>
+			<?php endif; ?>
+
+			<button
+				class="menu-toggle"
+				aria-controls="primary-menu"
+				aria-expanded="false"
+				<?php if ( _s_is_amp() ) : ?>
+					on="tap:AMP.setState( { siteNavigationMenu: { expanded: ! siteNavigationMenu.expanded } } )"
+					[aria-expanded]="siteNavigationMenu.expanded ? 'true' : 'false'"
+				<?php endif; ?>
+			>
+				<?php esc_html_e( 'Primary Menu', '_s' ); ?>
```

Comment:
I could not figure out what the button is for. Is it for the mobile menu?

Reply:
Yes. If you resize the window to the mobile breakpoint then this button appears.